coordinator/src/manyprover.rs

```diff
@@ -0,0 +1,275 @@
+//! This module contains everything to prove multiple blocks.
+use std::time::Instant;
+
+use log::{debug, error, info};
+use paladin::runtime::Runtime;
+use proof_gen::types::PlonkyProofIntern;
+
+use crate::benchmarking::{
+    BenchmarkingOutput, BenchmarkingOutputBuildError, BenchmarkingOutputError, BenchmarkingStats,
+};
+use crate::fetch::{fetch, FetchError};
+use crate::input::{ProveBlocksInput, TerminateOn};
+use crate::proofout::{ProofOutput, ProofOutputBuildError, ProofOutputError};
+
+//===========================================================================================
+// ManyProverError
+//===========================================================================================
+#[derive(Debug)]
+pub enum ManyProverError {
+    Fetch(FetchError),
+    Proof(anyhow::Error),
+    BenchmarkingOutputBuild(BenchmarkingOutputBuildError),
+    BenchmarkingOutput(BenchmarkingOutputError),
+    ProofOutBuildError(ProofOutputBuildError),
+    ProofOutError(ProofOutputError),
+}
+
+impl std::fmt::Display for ManyProverError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Debug::fmt(&self, f)
+    }
+}
+
+impl std::error::Error for ManyProverError {}
+
+//===========================================================================================
+// Proving function
+//===========================================================================================
+
+/// Proves many blocks
+pub async fn prove_blocks(
+    input: ProveBlocksInput,
+    runtime: &Runtime,
+) -> Result<(), ManyProverError> {
+    info!("Initializing to prove blocks...");
+
+    info!("{:#?}", input);
+
+    //=================================================================================
+    // Init & Setup
+    //=================================================================================
+
+    debug!("Preparing proof output...");
+    // get the proof
+    let proof_out = match &input.proof_output {
+        Some(proof_method) => match ProofOutput::from_method(proof_method) {
+            Ok(proof_out) => Some(proof_out),
+            Err(err) => {
+                error!("Failed to build proof out");
+                return Err(ManyProverError::ProofOutBuildError(err));
+            }
+        },
+        None => None,
```

Comment:
I find the `None => None` to be redundant. In cases like these, my preference is to use rust's functional features to **only** perform an action if the `Option` is not `None`, otherwise leave the existing `None` in-place. But this is a style choice, feel free to ignore. 

This is how I would've written it:
```
let proof_out = input.proof_output.as_ref().and_then(|proof_method| {
    ProofOutput::from_method(proof_method).map_err(|err| {
        error!("Failed to build proof out");
        ManyProverError::ProofOutBuildError(err)
    }).ok()
});
```